src/test/test-extract-word.c

```diff
@@ -341,6 +341,27 @@ static void test_extract_first_word(void) {
         assert_se(streq(t, "foo\\xbar"));
         free(t);
         assert_se(p == NULL);
+
+        p = "\\:";
+        assert_se(extract_first_word(&p, &t, ":", EXTRACT_UNESCAPE_SEPARATORS) == 1);
+        assert_se(streq(t, ":"));
+        free(t);
+        assert_se(p == NULL);
```

Comment:
maybe extend tests here so that the same calls without the new flag is called, and verify it yields the right (and different results)

Reply:
Done in https://github.com/systemd/systemd/pull/16321